src/app/pages/credentials/certificates-dash/forms/acmedns-form.component.ts

```diff
@@ -96,6 +36,57 @@ export class AcmednsFormComponent {
         this.queryCallOption = [['id', '=', rowId]]
         this.getRow.unsubscribe();
       })
+      this.ws.call('acme.dns.authenticator.authenticator_schemas', []).subscribe((schemas) => {
+        const fieldSet: any = [
+          {
+            name: 'Add DNS Authenticator',
+            label: true,
+            config:[
+              {
+                type : 'input',
+                name : 'name',
+                placeholder : helptext.authenticator_name_placeholder,
+                tooltip : helptext.authenticator_name_tooltip,
+                required: true,
+                validation : helptext.authenticator_name_validation,
+                parent: this
+              },
+              {
+                type : 'select',
+                name : 'authenticator',
+                placeholder : helptext.authenticator_provider_placeholder,
+                tooltip : helptext.authenticator_provider_tooltip,
+                options : [
+                ],
+                parent: this
+              }
+            ]
+          }]
+        for(let schema of schemas) {
+          fieldSet[0].config[1].options.push({label: schema.key, value: schema.key});
```

Comment:
That's not very clear what stand for `fieldSet[0].config[1]`. It may be more verbose if we use `FieldSets` class from https://github.com/truenas/webui/blob/master/src/app/pages/common/entity/entity-form/classes/field-sets.ts 

Then it may be something like:
```
const fieldSet = new FieldSets([...])
const authenticatorOptions = fieldSet.config("authenticator")
authenticatorOptions.options.push(...)
```

Just a thought. What do you think @ryousaf?

Reply:
Good idea! I'll try and get that done. Thanks!